crates/hulk/src/nao/double_buffered_reader.rs

```diff
@@ -0,0 +1,411 @@
+use std::{
+    io::{self, ErrorKind, Read},
+    mem::size_of,
+    os::unix::{io::AsRawFd, prelude::RawFd},
+};
+
+use super::hula::StateStorage;
+
+pub struct DoubleBufferedReader {
```

Comment:
What do you think of this with some generics in an extra crate? Or is it overkill?

Reply:
I wanted to define `pub struct DoubleBufferedReader<T>` but this does not work because I used `size_of::<T>()` in the type which is [currently unstable](https://github.com/rust-lang/rust/issues/60551). We could try but I thought, I don't need to spend effort generalizing for this single instantiation. But yes, if you have an idea, how we can implement it, I'd be happy to do so.

Reply:
Hmm.. i'm fine with either approach. One solution might be to drop `size_of::<T>()` and replace it with `const N: usize` or similar.
Another idea, can you just make an array of `T` and reinterpret it to a `[u8]` when reading from the socket?

Reply:
It is now generic over the read type. It is still located in the `hulk` crate because this is the single usage for now. We can extract it into a crate later.